src/fpd/enrichment.js

```diff
@@ -78,6 +98,31 @@ function removeUndef(obj) {
   return getDefinedParams(obj, Object.keys(obj))
 }
 
+// export async function tryToGetCdepLabel(cb = getCookieDeprecationLabel) {
+//   let cdep;
+//   if (isActivityAllowed(ACTIVITY_ACCESS_DEVICE, activityParams(MODULE_TYPE_PREBID, 'cdep'))) {
+//     return GreedyPromise.resolve(
+//       await cb(cdep)
+//     );
+//   }
+// }
+
+export async function tryToGetCdepLabel(cb = getCookieDeprecationLabel) {
+  return GreedyPromise.resolve(isActivityAllowed(ACTIVITY_ACCESS_DEVICE, activityParams(MODULE_TYPE_PREBID, 'cdep')) && await cb());
```

Comment:
I think the getCookieDeprecationLabel func would require that we wait for it to resolve right?  otherwise the label (if there is one) most likely wouldn’t get retrieved in time before the bid reqs go out.. (i'm not able to fetch it in time from my end at least).  Was wondering what your thoughts on this use of `await` here are?  Or if you have any other suggestions?

One thing I was thinking.. maybe right here a setTimeout would be good?  One that would only wait x amount of time for `navigator.cookieDeprecationLabel.getValue()` to resolve and if it doesn’t we can resolve manually (and not block the auction)?  I added in some arbitrary setTimeout code as an example, but of course that can be changed based on what you think.

Reply:
`getCookieDeprecationLabel` returns a promise; since this also returns a promise, `await` is not necessary. Unless I'm mistaken `await P; [...]`  is just syntactic sugar for `P.then(() => [...])` so in this case it does nothing.

Reply:
You're correct!  I think when I originally tried out your suggestion there was a bug on my end :joy: I've since tried it again and things are working as expected on my end :+1: